app/models/king.rb

```diff
@@ -1,4 +1,27 @@
 class King < Piece
-	def move
+
+	def valid_move?(start_point, end_point)
```

Comment:
Looking at this, it doesn't make sense to pass in the start_point.  If you are already dealing with a piece on the board then you already know the start point.  This is a bit advanced.  But look at the code we have here.  There are several things that should be methods in the Piece model so that all pieces can use them.  

1. getting the start position
2. checking the move is on the board
3. checking the start != end point

you should extract the pieces of code you are likely to re-use and put them in the Piece parent class.  Then you could rewrite this method like the following.

def valid_move?(end_point)
   return false unless self.on_board?(end_point)
   return false unless self.new_point?(end_point)
   (self. pos_x - end_x).abs <= 1 && (self. pos_y - end_y).abs <= 1
end